Add install case when users don't have any workflows

Originally from https://github.com/jonrohan/alfred-figma/issues/1

Tried to install a workflow (that uses alfred-link) via the cli `npm install --global alfred-figma` but kept getting this error:

```
> alfred-figma@0.2.0 postinstall /Users/misolori/.nvm/versions/node/v8.9.4/lib/node_modules/alfred-figma
> alfy-init

{ Error: Command failed: alfred-link
Error: Workflow directory `/Users/misolori/Library/Application Support/Alfred 3/Alfred.alfredpreferences/workflows` does not exist
    at pathExists.then.exists (/Users/misolori/.nvm/versions/node/v8.9.4/lib/node_modules/alfred-figma/node_modules/alfred-link/index.js:19:10)
    at <anonymous>


    at Promise.all.then.arr (/Users/misolori/.nvm/versions/node/v8.9.4/lib/node_modules/alfred-figma/node_modules/execa/index.js:236:11)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
  code: 1,
  killed: false,
  stdout: '',
  stderr: 'Error: Workflow directory `/Users/misolori/Library/Application Support/Alfred 3/Alfred.alfredpreferences/workflows` does not exist\n    at pathExists.then.exists (/Users/misolori/.nvm/versions/node/v8.9.4/lib/node_modules/alfred-figma/node_modules/alfred-link/index.js:19:10)\n    at <anonymous>\n',
  failed: true,
  signal: null,
  cmd: 'alfred-link',
  timedOut: false }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! alfred-figma@0.2.0 postinstall: `alfy-init`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the alfred-figma@0.2.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/misolori/.npm/_logs/2018-10-25T15_46_44_571Z-debug.log
```

Turns out that if you don't have any workflows installed there is no workflows directory in `Alfred.alfredpreferences`. This case should be added during install to avoid these errors.

--- a/index.js
+++ b/index.js
@@ -1,5 +1,6 @@
 'use strict';
 const path = require('path');
+const makeDir = require('make-dir');
 const pathExists = require('path-exists');
 const readPkgUp = require('read-pkg-up');
 const resolveAlfredPrefs = require('resolve-alfred-prefs');
@@ -14,13 +15,22 @@ sudoBlock();
 const getWorkflowDir = () => resolveAlfredPrefs().then(prefs => path.join(prefs, 'workflows'));
 
 const readPkg = workflowDir => pathExists(workflowDir)
-	.then(exists => {
-		if (!exists) {
-			throw new Error(`Workflow directory \`${workflowDir}\` does not exist`);
+	.then(workflowDirExists => {
+		if (workflowDirExists) {
+			return;
 		}
 
-		return readPkgUp();
-	});
+		return resolveAlfredPrefs()
+			.then(prefs => pathExists(prefs))
+			.then(prefsExists => {
+				if (!prefsExists) {
+					throw new Error('`Alfred.alfredpreferences` package does not exist');
+				}
+
+				return makeDir(workflowDir);
+			});
+	})
+	.then(() => readPkgUp());
 
 exports.link = opts => {
 	const options = Object.assign({

--- a/package.json
+++ b/package.json
@@ -53,5 +53,5 @@
 	},
 	"devDependencies": {
 		"xo": "^0.23.0"
-  }
+	}
 }

--- a/package.json
+++ b/package.json
@@ -42,6 +42,7 @@
   ],
   "dependencies": {
     "del": "^2.2.2",
+    "make-dir": "^1.3.0",
     "path-exists": "^3.0.0",
     "pify": "^2.3.0",
     "plist": "^2.0.1",

